src/components/Languages.vue

```diff
@@ -0,0 +1,55 @@
+<template>
+  <div>
+    <span v-if="!languages.trim()">(no languages)</span>
+    <a
+      v-else
+      href="#"
+      class="chip"
+      v-for="(language, index) in languagesWithColors"
+      :key="index"
+      :style="{ backgroundColor: language.backgroundColor, color: language.textColor }"
+    >
+      {{ language.text }}
+    </a>
+  </div>
+</template>
+
+<script>
+import colors from "@/util/colors.json";
+import { hexToRgb, setContrast } from "@/util/colors";
+
+export default {
+  name: "Language",
+  props: {
+    languages: {
+      type: String,
+      required: true
+    }
+  },
+  computed: {
+    languagesWithColors() {
+      return this.languages
+        .trim()
+        .split(" ")
+        .map(language => {
+          const backgroundColor = colors[language] || "#e4e4e4";
+          const textColor = colors[language] ? setContrast(hexToRgb(backgroundColor)) : "black";
```

Comment:
I have tested it and:
when `colors[language]` is falsy this code: `setContrast(hexToRgb("#e4e4e4"))` will produce `"black"`

So this code `const textColor = colors[language] ? setContrast(hexToRgb(backgroundColor)) : "black";` is same as: `const textColor = setContrast(hexToRgb(backgroundColor));`

My consideration now is to call those 2 functions for a result we already know.
What u think ?

Reply:
You're right.

Fixed.